shared/src/main/scala-2.11-2.12/scala/xml/ScalaVersionSpecific.scala

```diff
@@ -0,0 +1,22 @@
+package scala.xml
+
+import scala.collection.SeqLike
+import scala.collection.generic.CanBuildFrom
+
+object ScalaVersionSpecific {
```

Comment:
Shouldn't this be marked private?

Reply:
yes good point. i did this only in https://github.com/scala/scala-xml/pull/223, sorry for the confusion

Reply:
So we made these traits `private`.  This means people can't use it as a trait.  However, the traits are published, and they are revealed in the type signatures.

![scaladoc for scala.xml.NodeSeq](https://user-images.githubusercontent.com/358615/47116682-5cae7380-d230-11e8-9f1d-c4453c6c7ff5.png)

```
 :type -v <x y="z"/>.attributes
// Type signature
scala.xml.MetaData

// Internal Type structure
TypeRef(
  TypeSymbol(
    abstract class MetaData extends AbstractIterable[scala.xml.MetaData] 
                   with Iterable[scala.xml.MetaData] 
                   with Equality 
                   with ScalaVersionSpecificIterableSerializable[scala.xml.MetaData] 
                   with Serializable

  )
)
```

Cost of doing business or should we do something else?

Reply:
Fortunately, dropping such a trait from a future version will be caught by Mima:

```
> xml/mimaReportBinaryIssues
[info] scala-xml: found 13 potential binary incompatibilities while checking 
[info] against org.scala-lang.modules:scala-xml_2.13.0-M5:1.1.1
...
[error]  * the type hierarchy of class scala.xml.NodeSeq is different in 
[error]  current version. Missing types {scala.xml.ScalaVersionSpecificIterableSerializable}
[error]    filter with: ProblemFilters.exclude[MissingTypesProblem]("scala.xml.NodeSeq")